src/vs/editor/contrib/suggest/suggest.ts

```diff
@@ -152,11 +152,25 @@ export function ensureLowerCaseVariants(suggestion: CompletionItem) {
 }
 
 function createSuggestionResolver(provider: CompletionItemProvider, suggestion: CompletionItem, model: ITextModel, position: Position): (token: CancellationToken) => Promise<void> {
-	let cached: Promise<void>;
+	let cached: Promise<void> | undefined;
 	return (token) => {
 		if (!cached) {
 			if (typeof provider.resolveCompletionItem === 'function') {
-				cached = Promise.resolve(provider.resolveCompletionItem(model, position, suggestion, token)).then(value => { assign(suggestion, value); });
+				let hasResolved = false;
+				cached = Promise.resolve(provider.resolveCompletionItem(model, position, suggestion, token))
```

Comment:
@jrieken Another fix would be to only wrap the `provider.resolveCompletionItem(model, position, suggestion, token` in the cancellable promise instead of including the caching logic in it too